src/GeneratorState.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef GENSET_WHISPERPOWER_DDC_GENERATOR_STATE_HPP
+#define GENSET_WHISPERPOWER_DDC_GENERATOR_STATE_HPP
+
+#include <base/Time.hpp>
+#include <genset_whisperpower_ddc/GeneratorStatus.hpp>
+
+namespace genset_whisperpower_ddc {
+    struct GeneratorState {
+        base::Time time;
+
+        int rpm;
+        int udc_start_battery;
+        enum StatusA {
+            OVERALL_ALARM = 0x01,
+            ENGINE_TEMPERATURE_ALARM = 0x02,
+            PM_VOLTAGE_ALARM = 0x04,
+            OIL_PRESSURE_ALARM = 0x08,
+            EXHAUST_TEMPERATURE_ALARM = 0x10,
+            UAC1_ALARM = 0x20,
+            IAC1_ALARM = 0x40,
+            OIL_PRESSURE_HIGH_ALARM = 0x80
+        };
+
+        /** @meta bitfield /genset_whisperpower_ddc/GeneratorState/StatusA */
+        uint8_t statusA = 0;
```

Comment:
Simplify the structure by making a single uint32_t status and a single enum that is the merge of the three bytes.